packages/flutter_tools/lib/src/windows/build_windows.dart

```diff
@@ -94,10 +94,11 @@ Future<void> buildWindows(WindowsProject windowsProject, BuildInfo buildInfo, {
     status.stop();
   }
 
+  final String? binaryName = getCmakeExecutableName(windowsProject);
```

Comment:
This is the reland's fix. The original change assumed that the `pubspec.yaml`'s app name matched CMake's app name, but that might not be true if the user edited one file but not both.